policy.go

```diff
@@ -659,13 +741,32 @@ func (p *policyPlugin) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dn
 		}
 	}
 
-	// validate domain name (validation #1)
-	err = p.validate(ah, "")
-	if err != nil {
-		status = dns.RcodeServerFailure
-		goto Exit
-	}
+	if p.rule.keyword != "" {
+		result, err := p.evaluateExp(p.rule.params, p.rule.keyword, w, r)
+		if err != nil {
+			resolveFailed = true
+			goto Exit
+		}
+		if result == true {
```

Comment:
p.evaluateExp(..) should be pushed the PolycRule

=> p.policyRule[x].evaluate reply directly the action (typeAllow, typeRefuse or other ...)